### PR TITLE
Add view that can lock swipe-to-close gesture

### LINKED ISSUE #796: Horizontal Scrolling Issue with Tables
I can scroll to the right without any issue for tables, but whenever I try to scroll to the left, it wants to take me back to the post feed instead of scrolling the table.

I.E. https://www.reddit.com/r/arknights/comments/v8ad5w/module_details_lingering_echoes/

### PR DESCRIPTION
**Pull request by RSBat**
tl dr: Custom view that if necessary locks Slidr before the swipe starts
Full explanation below...

Fixes #796 

Slidr works by adding its own view in the hierarchy and listening to touch events in `onInterceptTouchEvent`. Once it detects movement in the correct direction, it returns `true` and handles all the events itself.

Adding scrollable view detection to Slidr would solve the problem, but it is not possible and would probably have performance impact.

Fortunately Slidr does not intercept the very first event, which is ACTION_DOWN, and it reaches scrollable view. So the scrollable view itself can decide if it should disallow the swipe.

This also has a performance benefit over `OnScrollChangedListener` because the listener is triggered for every scroll of every view even if the child we are interested in did not scroll. On the other hand `on(Intercept)TouchEvent` is triggered only when the view is touched.

There is a possibility that swipe won't be unlocked if view never receives ACTION_UP or ACTION_CANCEL. However the docs say nothing about the probability of this happening. Anyways, one possible solution is to post a runnable that will unlock swipe soon after locking.

### CODE REVIEW

**Comment by Docile-Alligator** [comment]
Thank you! Could you help resolve the conflicts?

**Comment by RSBat** [comment]
Fixed conflicts

**Comment by Docile-Alligator** [comment]
Thank you!

**Comment by RSBat** [comment]
Hey @Docile-Alligator , any plans for a new release soon?

**Comment by Docile-Alligator** [comment]
@RSBat Hi! I am extremely busy with my school work recently so no release until November 🥺.

**Comment by RSBat** [comment]
No problem, good luck with school!
